packages/react/src/experimental/OneDataCollection/utils.ts

```diff
@@ -0,0 +1,89 @@
+import { ReactNode } from "react"
+import { ColumnWidth } from "../OneTable/utils/sizes"
+import { VisualizationType } from "./visualizations"
+import { propertyRenderers } from "./visualizations/property"
+
+/**
+ * The definition of a renderer.
+ * Union type of all possible renderer definitions to ensure the value is the type related the `type`{ [RenderedType]: RendererFuncArgument }.
+ */
+export type RendererDefinition = {
+  [K in keyof typeof propertyRenderers]: {
+    type: K
+    value: Parameters<(typeof propertyRenderers)[K]>[0]
+  }
+}[keyof typeof propertyRenderers]
+
+export type PropertyDefinition<T> = {
+  label: string
+
+  /**
+   * Optional tooltip text. When provided, displays an info icon next to the header content
+   * that shows this text in a tooltip when hovered.
+   */
+  info?: string
+
+  /**
+   * The width of the column. If not provided, the width will be "auto"
+   */
+  width?: ColumnWidth
+
+  /**
+   * Function that extracts and formats the value from an item.
+   * Should return an object matching the expected args for the specified renderer type.
+   *
+   * Example usage:
+   * {
+   *   render: (item) => ({
+   *     type: "avatar",
+   *     value: {
+   *       type: "person",
+   *       firstName: item.firstName,
+   *       lastName: item.lastName,
+   *     }
+   *   })
+   * }
+   */
+  render: (item: T) => RendererDefinition | string | number | undefined
+}
+
+/**
+ * Renders a value for a given item and property definition.
+ * Used by both table and card visualizations to ensure consistent rendering.
+ */
+export const renderValue = <RecordType>(
+  item: RecordType,
+  property: PropertyDefinition<RecordType>,
+  visualization: VisualizationType
+): ReactNode => {
+  const renderIsRendererDefinition = (
+    renderDef: RendererDefinition | string | number | undefined
+  ): renderDef is RendererDefinition => {
+    return renderDef !== undefined && typeof renderDef === "object"
+  }
```

Comment:
let's move it out of the function to avoid recreation of the function on every call. It should be safe since you don't work with hooks here